app/routes.py

```diff
@@ -1,2 +1,145 @@
+    view_tasks = [task.to_json() for task in tasks if tasks]
+    return jsonify(view_tasks)
+
+@tasks_bp.route("/<task_id>", methods=["GET"], strict_slashes=False)
+def view_task(task_id):
+    task = Task.query.get_or_404(task_id)
+    return jsonify(task = task.to_json())
+
+@tasks_bp.route("/<task_id>", methods=["PUT"], strict_slashes=False)
+def update_task(task_id):
+    task = Task.query.get_or_404(task_id)
+    updated_task = request.get_json()
+    if not updated_task:
+        return jsonify(None), 404
+    task.title = updated_task['title']
+    task.description = updated_task["description"]
+    task.completed_at = updated_task["completed_at"]
+    db.session.commit()
+    send_notification_to_slack(f"Task {task_id} updated.")
+    return jsonify(task=task.to_json())
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"], strict_slashes=False)
+def delete_task(task_id):
+    task = Task.query.get_or_404(task_id)
+    db.session.delete(task)
+    db.session.commit()
+    send_notification_to_slack(f"Task {task_id} deleted.")
+    return jsonify(details = f'Task {task.task_id} "{task.title}" successfully deleted')
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"], strict_slashes=False)
+def mark_incomplete(task_id):
+    send_notification_to_slack(f"Task {task_id} undone.")
+    return mark_task(task_id, False) 
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"], strict_slashes=False)
+def mark_complete(task_id):
+    send_notification_to_slack(f"Task {task_id} completed!")
+    return mark_task(task_id, True)
+
+def mark_task(task_id, completed):
+    task = Task.query.get_or_404(task_id)
+    status = datetime.utcnow() if completed else None
+    task.completed_at = status
+    db.session.commit()
+    return jsonify(task=task.to_json())
+
+######################## GOAL ROUTES ######################## 
+
+@goals_bp.route("", methods=["POST"], strict_slashes=False)
+def create_new_goal():
+    request_body = request.get_json()
+    if len(request_body) < 1:
+        return jsonify(details= f'Invalid data'), 400
+    new_goal = Goal(title = request_body["title"])
+    db.session.add(new_goal)
+    db.session.commit()
+    return jsonify(goal=new_goal.to_json()), 201
+
+@goals_bp.route("", methods=["GET"], strict_slashes=False)
+def get_all_goals():
+    query_param_value = request.args.get("sort")
+    goals = Goal.sort(query_param_value)
+    view_goals= [goal.to_json() for goal in goals if goals]
+    return jsonify(view_goals)
+
+@goals_bp.route("/<goal_id>", methods=["GET"], strict_slashes=False)
+def view_goal(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+    return jsonify(goal=goal.to_json())
+
+@goals_bp.route("/<goal_id>", methods=["PUT"], strict_slashes=False)
+def update_goal(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+    updated_goal = request.get_json()
+    if not updated_goal:
+        return jsonify(None), 404
+    goal.title = updated_goal["title"]
```

Comment:
As you mentioned in the comments in Learn, you've handled invalid request data. I think what I see is that you check whether the request body is empty or not. Consider what would happen if the request body only had information for an updated description but not at updated title. How might you handle this situation?